### PR TITLE
Note the type when unable to drop values in compile time

### PR DESCRIPTION
**Pull request by fee1-dead**

### CODE REVIEW

File: compiler/rustc_const_eval/src/transform/check_consts/ops.rs [hunk 1]
```diff
@@ -438,6 +439,7 @@ impl<'tcx> NonConstOp<'tcx> for LiveDrop {
             "destructors cannot be evaluated at compile-time"
```
Comment (oli-obk) [inline compiler/rustc_const_eval/src/transform/check_consts/ops.rs:439]: Considering that we'll end up allowing destructors of some types, maybe just change the main message to say "destructor of type `Foo` can't be..."

**Review by oli-obk — CHANGES_REQUESTED** [review]

File: compiler/rustc_const_eval/src/transform/check_consts/ops.rs (hunk 1 above)
Comment (fee1-dead) [reply compiler/rustc_const_eval/src/transform/check_consts/ops.rs]: done

**Pull request merged**